lib/org/openpsa/user/handler/person/account.php

```diff
@@ -83,7 +83,9 @@ public function _handler_welcome_email(string $guid)
         $person = new midcom_db_person($guid);
         $accounthelper = new org_openpsa_user_accounthelper($person);
         $accounthelper->welcome_email();
-
+        if ($accounthelper->welcome_email()) {
+            midcom::get()->uimessages->add($this->_l10n->get('org.openpsa.user'), $this->_l10n->get('password reset and mail sent'));
+        }
```

Comment:
Can you also add an else case and show the errstr in case something goes wrong?

Reply:
Added

Reply:
The error message does not need to show the email address. Instead, it should show `errstr` from accounthelper. Also, I think it would be better if the type was error instead of warning